Makefile

```diff
@@ -34,9 +34,17 @@ TOPDIR=${CURDIR}
 TESTDIR=${TOPDIR}/tests
 include ${TESTDIR}/Makefile
 
-sources:
+.PHONY: uncommitted-check
+uncommitted-check:
+	@if ! git update-index --refresh --unmerged || \
+	    ! git diff-index --name-only --exit-code HEAD; then \
+	      echo "*** ERROR: Uncommitted changes in above files"; exit 1; fi
+
+.PHONY: sources
+sources: uncommitted-check
```

Comment:
Would it make sense to be able to override this error? Running "make sources" from a local dirty repo doesn't like something we should make impossible

Reply:
Sure, let's make have `release-sources` and `scratch-sources` targets. The former will have stricter checks, while the latter will bundle just about anything.